lib/src/http/managers/interaction_manager.dart

```diff
@@ -313,17 +349,25 @@ class InteractionManager {
         jsonPayload: jsonEncode(payload),
         files: files,
         applyGlobalRateLimit: false,
+        queryParameters: withResponse != null ? {'with_response': withResponse.toString()} : {},
       );
     } else {
       request = BasicRequest(
         route,
         method: 'POST',
         body: jsonEncode(builder.build()),
         applyGlobalRateLimit: false,
+        queryParameters: withResponse != null ? {'with_response': withResponse.toString()} : {},
       );
     }
 
-    await client.httpHandler.executeSafe(request);
+    final response = await client.httpHandler.executeSafe(request);
+
+    if (withResponse == null || withResponse == false) {
+      return null;
+    }
+
```

Comment:
Needs client.updateCacheWith call here (for the nested Message object). This will also require updating client.updateCacheWith to handle the appropriate data structures.

Reply:
Good catch, thanks.